src/player.js

```diff
@@ -1210,6 +1216,55 @@ export default class Player extends FakeEventTarget {
 
   // </editor-fold>
 
+  // <editor-fold desc="PIP API">
+
+  /**
+   * Request the player to enter picture in picture mode
+   * @public
+   * @returns {void}
+   */
+  enterPictureInPicture(): void {
+    if (!this._isInPip) {
+      this._engine.enterPip();
+    }
+  }
+
+  /**
+   * Request the player to exit picture in picture mode
+   * @public
+   * @returns {void}
+   */
+  exitPictureInPicture(): void {
+    if (this._isInPip) {
+      this._engine.exitPip();
+    }
+  }
+
+  /**
+   * Check if the player is in picture in picture mode
+   * @public
+   * @return {boolean} if the player is in picture in picture mode or not
+   */
+  isInPictureInPicture(): boolean {
+    return this._isInPip;
+  }
+
+  /**
+   * Check if picture in picture supported in this environment
+   * @public
+   * @return {boolean} if the picture in picture feature is supported in this environment
+   */
+  isPictureInPictureSupported(): boolean {
+    return this._engine.isPipSupported();
+  }
+
+  _handlePipEvent(event: FakeEvent): void {
+    this._isInPip = !this._isInPip;
```

Comment:
Don't change the state on the event itself, this is very dangerous.
If someone else will register to this event before this listener it will get a wrong state.
Update the state in the method which triggers the event.

Reply:
not relevant (moved to the engine)